File: app/src/main/java/com/example/pruebaconceptonavigationmanager/description/DescriptionActivity.java
package com.example.pruebaconceptonavigationmanager.description;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.example.navigation.stepsEngine.flow.rules.base.Rule;
import com.example.navigation.stepsEngine.payment.FlowState;
import com.example.pruebaconceptonavigationmanager.R;
import com.example.navigation.action.Action;
import com.example.navigation.action.Field;

import java.util.List;

public class DescriptionActivity extends AppCompatActivity implements Action {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_payment_description);
    }



    @Override
    public String getName() {
        return null;
    }

    @Override
    public List<Field> getFields() {
        return null;
    }

    @Override
    public void resolveUnfullfiledRule(Rule unfulfilledRule) {

    }

    @Override
    public void setField(String idField, Object value) {

    }

    @Override
    public void execute(List<Field> newFields) {

    }

    @Override
    public void backStepState(List<Field> fields, FlowState paymentFlowState) {

    }
}

File: app/src/main/java/com/example/pruebaconceptonavigationmanager/description/DescriptionActivity.kt
package com.example.pruebaconceptonavigationmanager.description

import android.os.Bundle
import android.widget.Toast
import com.example.navigation.action.FlowManager
import com.example.navigation.stepsEngine.field.DescriptionField
import com.example.navigation.stepsEngine.field.Field
import com.example.navigation.stepsEngine.field.FieldName
import com.example.navigation.stepsEngine.flow.rules.actionValidation.ActionValidation
import com.example.pruebaconceptonavigationmanager.R
import com.example.pruebaconceptonavigationmanager.actions.ActionAbstractActivity
import com.example.pruebaconceptonavigationmanager.actions.ActionName
import kotlinx.android.synthetic.main.activity_payment_description.*

class DescriptionActivity : ActionAbstractActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_payment_description)

        button.setOnClickListener {
            confirmInputs()
        }
    }

    private fun confirmInputs() {
        setField(FieldName.DESCRIPTION, txtDescription.text.toString())

        FlowManager.i!!.next(this)
    }

    override val name: String
        get() = ActionName.DESCRIPTION


    override fun resolveUnfullfiledRule(unfulfilledRule: ActionValidation) {
        when (unfulfilledRule.id) {
            ActionValidation.Codes.DESCRIPTION_TOO_LARGE -> {
                Toast.makeText(applicationContext, "Descripcion demasiado larga", Toast.LENGTH_LONG).show()
            }
        }

    }


}